planner/core/preprocess.go

```diff
@@ -616,7 +624,7 @@ func (p *preprocessor) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
 			p.err = err
 			return
 		}
-		isPrimary, err := checkColumnOptions(colDef.Options)
+		isPrimary, err := checkColumnOptions(stmt.TemporaryKeyword != ast.TemporaryNone, colDef.Options)
```

Comment:
```
tidb> create temporary table t(a bigint key auto_random);
ERROR 8006 (HY000): `auto_random` is unsupported on temporary tables.
```
The current PR forbids 'auto_random' with 'MySQL/local temporary' table(although it is a no-op keyword when `tidb_enable_noop_functions=true`).

Please confirm if it is expected(I think it should be fine), if yes, please add a test case for it.